service-base/src/main/java/org/eclipse/hono/service/management/device/AutoProvisioningService.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service.management.device;
+
+import java.security.cert.X509Certificate;
+
+import org.eclipse.hono.service.management.OperationResult;
+
+import io.opentracing.Span;
+import io.vertx.core.Future;
+
+/**
+ * A service for automatically provisioning devices.
+ *
+ * @see <a href="https://www.eclipse.org/hono/docs/dev/concepts/device-provisioning/#automatic-device-provisioning">
+ *      Automatic Device Provisioning</a>
+ */
+public interface AutoProvisioningService {
```

Comment:
this interface doesn't seem to be used anywhere except for the `AutoProvisioningEnabledDeviceBackend` ... 
Is it needed for anything else?

Reply:
I think an `AutoProvisioningService` that is not a `DeviceBackend` could be possible. But tbh, it just follows the very modular structure of interfaces within the Device Registry. Which to my understanding should provide great flexibility for *other* implementations, while their usage cannot be fully derived from the implementation of the example device registry.

Reply:
In that case, why would anybody want to implement just `AutoProvisioningService` if this interface is not used anywhere?

Reply:
I removed the interface.